src/evm/engine_db/simulation_db.rs

```diff
@@ -322,11 +314,11 @@ where
     }
 }
 
-impl<M: Middleware + Debug> DatabaseRef for SimulationDB<M>
+impl<P: Provider + Debug> DatabaseRef for SimulationDB<P>
 where
-    M::Error: std::fmt::Debug,
+    P: Provider + Send + Sync + 'static,
```

Comment:
Better to keep everything together, also `Provider` is specified twice

```suggestion
impl<P> DatabaseRef for SimulationDB<P>
where
    P: Provider + Debug + Send + Sync + 'static,
```

And same question as above: what forces us to make it `'static`?